user_service/repo/user.go

```diff
@@ -2,13 +2,16 @@ package repo
 
 import (
 	"database/sql"
+	"errors"
 	"fmt"
 
 	"user_service/models"
 )
 
 type IUserRepo interface {
 	GetUserByEmail(email string) (models.User, error)
+	InsertUser(u models.SignupRequest) error
+	VerifyUser(email string) error
```

Comment:
VerifyUser should not be part of user repo, user repo should be specific to DB operations, something like:
```
UpdateUser(user models.User) error
or
UpdateUserByID(id int, user models.User) error
```